capsules/src/max17205.rs

```diff
@@ -291,24 +376,30 @@ impl<'a> MAX17205Driver<'a> {
 }
 
 impl<'a> MAX17205Client for MAX17205Driver<'a> {
-    fn status(&self, status: u16) {
-        self.callback.get().map(|mut cb| cb.schedule(1, status as usize, 0));
+    fn status(&self, status: u16, error: ReturnCode) {
+        self.callback.get().map(|mut cb| cb.schedule(From::from(error), status as usize, 0));
     }
 
-    fn state_of_charge(&self, percent: u16, capacity: u16, full_capacity: u16) {
+    fn state_of_charge(&self, percent: u16, capacity: u16, full_capacity: u16, error: ReturnCode) {
         self.callback.get().map(|mut cb| {
-            cb.schedule(2,
+            cb.schedule(From::from(error),
                         percent as usize,
                         (capacity as usize) << 16 | (full_capacity as usize));
         });
     }
 
-    fn voltage_current(&self, voltage: u16, current: u16) {
-        self.callback.get().map(|mut cb| cb.schedule(3, voltage as usize, current as usize));
+    fn voltage_current(&self, voltage: u16, current: u16, error: ReturnCode) {
+        self.callback
+            .get()
+            .map(|mut cb| cb.schedule(From::from(error), voltage as usize, current as usize));
     }
 
-    fn coulomb(&self, coulomb: u16) {
-        self.callback.get().map(|mut cb| cb.schedule(4, coulomb as usize, 0));
+    fn coulomb(&self, coulomb: u16, error: ReturnCode) {
+        self.callback.get().map(|mut cb| cb.schedule(From::from(error), coulomb as usize, 0));
+    }
+
+    fn romid(&self, error: ReturnCode) {
+        self.callback.get().map(|mut cb| cb.schedule(From::from(error), 0, 0));
```

Comment:
Instead of using `allow`, you could just use the remaining two arguments to fit a u64 (by splitting it into two `u32`s):

```rust
fn romid(&self, rid: u64, error: ReturnCode) {
    self.callback.get.map(|mut cb| cb.schedule(From::from(error), (rid & 0xffffffff) as u32, (rid >> 32) as u32);
}
```

Reply:
I had thought about this, but I wasn't sure if it was actually cleaner or not? But I suppose avoiding the allow is good?

Reply:
I _think_ so

Reply:
I agree that just returning data directly in the subscribe callback is cleaner where possible. I hadn't even thought about splitting it over two arguments.